syntax = "proto3";

service SchedulerToTrainer {
    rpc QueryStats (QueryStatsRequest) returns (QueryStatsResponse) {};
    rpc SaveModel (SaveModelRequest) returns (SaveModelResponse) {};
}

message QueryStatsRequest {

}

message QueryStatsResponse {
    uint32 finished_iterations = 1;
    double iteration_time = 2;
}

message SaveModelRequest {

}

message SaveModelResponse {
    bool success = 1;
}